src/main/java/org/opentripplanner/routing/graphfinder/PlaceFinderTraverseVisitor.java

```diff
@@ -176,8 +179,22 @@ private void handleStop(RegularStop stop, double distance) {
       !seenStops.contains(stop.getId()) &&
       (filterByModes == null || stopHasPatternsWithMode(stop, filterByModes))
     ) {
-      placesFound.add(new PlaceAtDistance(stop, distance));
-      seenStops.add(stop.getId());
+      if (
+        includeStopsAndStations &&
+        stop.getParentStation() != null &&
+        !seenStops.contains(stop.getParentStation().getId()) &&
+        !seenStops.contains(stop.getId())
+      ) {
+        seenStops.add(stop.getParentStation().getId());
+        placesFound.add(new PlaceAtDistance(stop.getParentStation(), distance));
```

Comment:
Hmm this distance here is a bit questionable as it's the distance to a random (or maybe closest stop) of a station instead to some the location of the station (I think they always have lat and lon, not sure). However, since we are finding this distance by travelling through the street network and the stations themselves are not connected to the street network, maybe this is the best we can do.